registry_no_code_module_integration_test.go

```diff
@@ -0,0 +1,327 @@
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+			require.Error(t, err)
+			require.Nil(t, noCodeModule)
+		})
+		t.Run("with follow_latest_version not present", func(t *testing.T) {
+			options := RegistryNoCodeModuleCreateOptions{
+				Enabled:        Bool(true),
+				RegistryModule: registryModuleTest,
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+			require.Error(t, err)
+			require.Nil(t, noCodeModule)
+		})
+		t.Run("with version_pin and follow_latest_version present", func(t *testing.T) {
+			options := RegistryNoCodeModuleCreateOptions{
+				Enabled:             Bool(true),
+				FollowLatestVersion: Bool(true),
+				VersionPin:          String("1.2.3"),
+				RegistryModule:      registryModuleTest,
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+			require.Error(t, err)
+			require.Nil(t, noCodeModule)
+		})
+		t.Run("with registry module not present", func(t *testing.T) {
+			options := RegistryNoCodeModuleCreateOptions{
+				Enabled:             Bool(true),
+				FollowLatestVersion: Bool(true),
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+			require.Error(t, err)
+			require.Nil(t, noCodeModule)
+		})
+
+		t.Run("with invalid registry module", func(t *testing.T) {
+			options := RegistryNoCodeModuleCreateOptions{
+				FollowLatestVersion: Bool(true),
+				Enabled:             Bool(true),
+				RegistryModule:      &RegistryModule{ID: "invalid"},
+			}
+
+			noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+			assert.Error(t, err)
+			assert.Nil(t, noCodeModule)
+		})
+	})
+
+	t.Run("with variable options", func(t *testing.T) {
+		registryModuleTest, registryModuleTestCleanup := createRegistryModule(t, client, orgTest, PrivateRegistry)
+		defer registryModuleTestCleanup()
+
+		options := RegistryNoCodeModuleCreateOptions{
+			FollowLatestVersion: Bool(true),
+			Enabled:             Bool(true),
+			RegistryModule:      registryModuleTest,
+			VariableOptions: []*NoCodeVariableOption{
+				{
+					VariableName: "var1",
+					VariableType: "string",
+					Options:      []string{"option1", "option2"},
+				},
+				{
+					VariableName: "my_var",
+					VariableType: "string",
+					Options:      []string{"my_option1", "my_option2"},
+				},
+			},
+		}
+
+		noCodeModule, err := client.RegistryNoCodeModules.Create(ctx, orgTest.Name, options)
+		require.NoError(t, err)
+		assert.NotEmpty(t, noCodeModule.ID)
+		require.NotEmpty(t, noCodeModule.Organization)
+		require.NotEmpty(t, noCodeModule.RegistryModule)
+		assert.Equal(t, *options.FollowLatestVersion, noCodeModule.FollowLatestVersion)
+		assert.Equal(t, *options.Enabled, noCodeModule.Enabled)
+		assert.Equal(t, orgTest.Name, noCodeModule.Organization.Name)
+		assert.Equal(t, noCodeModule.RegistryModule.ID, noCodeModule.RegistryModule.ID)
```

Comment:
replace assertion that compares noCodeModule.RegistryModule.ID to itself
```suggestion
		assert.Equal(t, registryModuleTest.ID, noCodeModule.RegistryModule.ID)
```